lib/puppet/util/run_mode.rb

```diff
@@ -114,8 +130,32 @@ def log_dir
         which_dir(File.join(windows_common_base("puppet/var/log")), "~/.puppetlabs/var/log")
       end
 
+      def pkg_config_path
+        nil
+      end
+
+      def gem_cmd
+        if (puppet_dir = ENV.fetch('PUPPET_DIR', nil))
+          File.join(puppet_dir.to_s, 'bin', 'gem.bat')
+        else
+          File.join(Gem.default_bindir, 'gem.bat')
+        end
+      end
+
+      def common_module_dir
+        "#{installdir}/puppet/modules" if installdir
```

Comment:
I kept this as it used to be, but I was wondering if this is better
```suggestion
        File.join(installdir, 'puppet', 'modules') if installdir
```